src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -60,6 +61,24 @@ public PythonEngineVersion Engine
                 }
             }
         }
+        /// <summary>
+        /// Return the user selected python engine enum.
+        /// </summary>
+        public PythonEngineVersion CachedEngine
```

Comment:
This does not seem like something which needs to be exposed as a public property... it instead seems like something that is an internal implementation detail.

Reply:
making this internal makes the initial dropdown value to be blank, this is accessed directly by the dropdown

Reply:
Yup. I believe direct binding on WPF assumes the property to be public..

Reply:
I understand that, but why are we binding to this at all? Why not continue binding to the previous engine property. It's Adding public properties to a NodeModel type (its API) is not something that should be done without a good reason.

Reply:
It sounds to me, like you don't want to use a binding on the NodeModel - this is purely a `View` level property correct? I think you should bind to the view or viewModel for this property, and add this property there.

It feels strange to me to put a public property on the pythonNode which has NO impact on execution, only on view, meaning if the view was not present this property is useless, and also confusing.